### PR TITLE
[ExceptionList] Change warning status items from orange to yellow

### LINKED ISSUE #804: [Exception list] Status colors are hard to distinguish
It’s hard to distinguish orange from red in exception list messages.
screen shot 2018-05-31 at 5 27 28 pm

Let’s change the color of the warning status icon and title to use yellow, dark.
screen shot 2018-05-31 at 5 59 15 pm

Related to https://github.com/Shopify/polaris-rails/issues/1391

### PR DESCRIPTION
**Pull request by chloerice**
### WHY are these changes introduced?

Fixes #804 

### WHAT is this pull request doing?

Updates the font, icon, and bullet colors of warning status `ExceptionList` items from dark orange to dark yellow.

| Before | After |
|---|---|
|screen shot 2019-01-04 at 11 51 55 am|screen shot 2019-01-04 at 11 50 59 am|

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {
  Page,
  ResourceList,
  Card,
  ExceptionList,
  TextStyle,
} from '@shopify/polaris';

interface State {
  selectedItems: any[];
}

export default class Playground extends React.Component<{}, State> {
  state: State = {
    selectedItems: [],
  };

  handleSelectionChange = (selectedItems: any[]) => {
    this.setState({selectedItems});
  };

  renderItem = (item, id) => {
    const {orderId, address} = item;
    const exceptions =
      id === 256
        ? [
            {
              status: 'critical',
              icon: 'risk',
              title: 'High fraud risk',
            },
            {
              status: 'warning',
              icon: 'alert',
              title: 'No available inventory',
              description: '-2 in stock',
            },
            {
              icon: 'notes',
              title: 'This customer is awesome',
            },
          ]
        : [];
    return (
      <ResourceList.Item
        id={id}
        accessibilityLabel={`View details for ${name}`}
      >
        <h3>
          <TextStyle variation="strong">{orderId}</TextStyle>
        </h3>
        <div>{address}</div>
        <div style={{marginLeft: '-4px', marginTop: '4px'}}>
          <ExceptionList items={exceptions} />
        </div>
      </ResourceList.Item>
    );
  };

  render() {
    const resourceName = {
      singular: 'order',
      plural: 'orders',
    };

    const items = [
      {
        id: 341,
        orderId: '120341',
        address: 'Decatur, USA',
      },
      {
        id: 256,
        orderId: '120256',
        name: 'Ellen Ochoa',
        address: 'Los Angeles, USA',
      },
    ];

    const promotedBulkActions = [
      {
        content: 'Edit orders',
        onAction: () => console.log('Todo: implement bulk edit'),
      },
    ];

    const bulkActions = [
      {
        content: 'Add tags',
        onAction: () => console.log('Todo: implement bulk add tags'),
      },
      {
        content: 'Remove tags',
        onAction: () => console.log('Todo: implement bulk remove tags'),
      },
      {
        content: 'Archive orders',
        onAction: () => console.log('Todo: implement bulk archive'),
      },
    ];

    return (
      <Page
        title="Orders"
        primaryAction={{content: 'View Examples', url: '/examples'}}
      >
        <Card>
          <ResourceList
            resourceName={resourceName}
            items={items}
            renderItem={this.renderItem}
            selectedItems={this.state.selectedItems}
            onSelectionChange={this.handleSelectionChange}
            promotedBulkActions={promotedBulkActions}
            bulkActions={bulkActions}
          />
        </Card>
      </Page>
    );
  }
}
```

</details>

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)

### CODE REVIEW

**Comment by chloerice** [comment]
> It looks like in the original issue, the comparison was for a white background. Here, against the light gray background, the yellow technically doesn't pass (it's 4.12:1) for normal sized text (which I should have caught since that's what's used in the actual example in the style guide, doh).

Oy 😣 I've made the Playground to be more realistic to make it easier to see the various states/conditions the exception list component can be in. It's typically used inside of a `ResourceList.Item` or inside of a `Banner` (for listing multiple form validation errors at the top of a detail page). The hover, selected, and focus states of `ResourceList.Item` likely give even worse contrast results than the gray background. @dpersing is that a Chrome extension you're using to inspect the contrast ratio?

**Comment by sarahill** [comment]
Ok, I messed around with this a bit. I tried using the darker yellows and yellow text but those get muddy and really dark. Reducing the brightness of `yellow dark` looks the best and passes contrast standards. Thoughts?

image

**Comment by dpersing** [comment]
> Reducing the brightness of yellow dark looks the best and passes contrast standards.

@sarahill That works for me. Would it be possible to update the hex in the UI kit and style guide to match `#8A6166`?

**Comment by sarahill** [comment]
@dpersing I was wondering if we should just update the hex value for `yellow dark`. It will affect icons and some text in the UI but I don't think it will be a glaringly obvious change that will impact negatively.

@chloerice can you update the value in `polaris-react`? I'll create issues for the `UI Kit`, `Style guide` and `Polaris Tokens`.

image

**Comment by chloerice** [comment]
> @chloerice It's built into Chrome now (was in Canary for a long time)! If you click on a text color swatch in the CSS window it does its best to calculate the contrast based on what the background color appears to be.

Ahhh okay! (I was looking for it in the "Accessibility" section of the dev tools.)

**Comment by chloerice** [comment]
> @chloerice can you update the value in polaris-react? I'll create issues for the UI Kit, Style guide and Polaris Tokens.

@sarahill On it!!

**Comment by sarahill** [comment]
@chloerice Ah that makes sense! I'm on it 👊

**Comment by sarahill** [comment]
https://github.com/Shopify/polaris-tokens/pull/44

**Comment by chloerice** [comment]
> Ok, the update to color-yellow-dark has been merged in polaris-tokens Shopify/polaris-tokens#44

Awesome!! 👏 I've added an upgrade to the new release of `polaris-tokens` to this PR.

**Comment by dpersing** [comment]
This looks great to me! Thank you so much, @sarahill and @chloerice!

File: UNRELEASED.md [hunk 1]
```diff
@@ -34,6 +35,8 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Dependency upgrades
 
+- Upgraded `@shopify/polaris-tokens` to v2.1.1 ([#813](https://github.com/Shopify/polaris-react/pull/813))
```
Comment (dpersing) [inline UNRELEASED.md]: 👍

**Pull request merged**